### ISSUE TITLE
Vali Loss: nan Test Loss: nan

### ISSUE THREAD

**Issue by ywwsc** [zhouhaoyi/Informer2020#342]
我自己的数据跑得出的loss是nan，这是为什么？我的数据不含nan或全0。

```
Args in experiment:
Namespace(activation='gelu', attn='prob', batch_size=16, c_out=1955, checkpoints='./checkpoints/', cols=None, d_ff=512, d_layers=1, d_model=2048, data='raw_static_wenzhou_dataset_201401', data_path='raw_static_wenzhou_dataset_201401.json', dec_in=1955, des='test', detail_freq='h', devices='0,1,2,3', distil=True, do_predict=False, dropout=0.05, e_layers=2, embed='timeF', enc_in=1955, factor=5, features='M', freq='h', gpu=0, inverse=False, itr=2, label_len=24, learning_rate=0.0001, loss='mse', lradj='type1', mix=True, model='informer', n_heads=1955, num_workers=0, output_attention=False, padding=0, patience=3, pred_len=12, root_path='./data/', s_layers=[3, 2, 1], seq_len=48, target='_', train_epochs=4, use_amp=False, use_gpu=True, use_multi_gpu=False)
Use GPU: cuda:0
>>>>>>>start training : informer_raw_static_wenzhou_dataset_201401_ftM_sl48_ll24_pl12_dm2048_nh1955_el2_dl1_df512_atprob_fc5_ebtimeF_dtTrue_mxTrue_test_0>>>>>>>>>>>>>>>>>>>>>>>>>>
train 461
val 65
test 137
Epoch: 1 cost time: 7.115729093551636
Epoch: 1, Steps: 28 | Train Loss: 0.9460290 Vali Loss: nan Test Loss: nan
Validation loss decreased (inf --> nan).  Saving model ...
```

debug发现在vali中的求TokenEmbedding时所用的conv1d得到的值含有nan或者inf，求大神解答。

**Comment by Yuuuuuzw** [comment]
请问解决了吗

**Comment by Yuuuuuzw** [comment]
> 我自己的数据跑得出的loss是nan，这是为什么？我的数据不含nan或全0。
> 
> ```
> Args in experiment:
> Namespace(activation='gelu', attn='prob', batch_size=16, c_out=1955, checkpoints='./checkpoints/', cols=None, d_ff=512, d_layers=1, d_model=2048, data='raw_static_wenzhou_dataset_201401', data_path='raw_static_wenzhou_dataset_201401.json', dec_in=1955, des='test', detail_freq='h', devices='0,1,2,3', distil=True, do_predict=False, dropout=0.05, e_layers=2, embed='timeF', enc_in=1955, factor=5, features='M', freq='h', gpu=0, inverse=False, itr=2, label_len=24, learning_rate=0.0001, loss='mse', lradj='type1', mix=True, model='informer', n_heads=1955, num_workers=0, output_attention=False, padding=0, patience=3, pred_len=12, root_path='./data/', s_layers=[3, 2, 1], seq_len=48, target='_', train_epochs=4, use_amp=False, use_gpu=True, use_multi_gpu=False)
> Use GPU: cuda:0
> >>>>>>>start training : informer_raw_static_wenzhou_dataset_201401_ftM_sl48_ll24_pl12_dm2048_nh1955_el2_dl1_df512_atprob_fc5_ebtimeF_dtTrue_mxTrue_test_0>>>>>>>>>>>>>>>>>>>>>>>>>>
> train 461
> val 65
> test 137
> Epoch: 1 cost time: 7.115729093551636
> Epoch: 1, Steps: 28 | Train Loss: 0.9460290 Vali Loss: nan Test Loss: nan
> Validation loss decreased (inf --> nan).  Saving model ...
> ```
> 
> debug发现在vali中的求TokenEmbedding时所用的conv1d得到的值含有nan或者inf，求大神解答。

请问解决了吗

**Comment by ywwsc** [comment]
> 请问解决了吗

解决了，在我的数据里发现脏数据，随时间变化全是0的数据，去掉改数据得以解决

**Comment by Yuuuuuzw** [comment]
> 
谢谢谢谢，我去检查下我的数据
